lib/services/graph-progress.js

```diff
@@ -0,0 +1,230 @@
+// Copyright © 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+'use strict';
+
+module.exports = graphProgressServiceFactory;
+
+graphProgressServiceFactory.$provide = 'Services.GraphProgress';
+graphProgressServiceFactory.$inject = [
+    '_',
+    'Protocol.Events',
+    'Services.Waterline',
+    'Logger',
+    'Constants',
+    'Promise',
+    'Assert',
+    'GraphProgress'
+];
+
+function graphProgressServiceFactory(
+    _,
+    eventsProtocol,
+    waterline,
+    Logger,
+    Constants,
+    Promise,
+    assert,
+    GraphProgress
+) {
+    var logger = Logger.initialize(graphProgressServiceFactory);
+
+    function GraphProgressService() {
+    }
+
+    /**
+     * Publish a graph started progress event over AMQP.
+     *
+     * @param {Object} graph
+     * @returns {Promise}
+     * @memberOf GraphProgressService
+     */
+    GraphProgressService.prototype.publishGraphStarted = function(graph) {
```

Comment:
How you make sure the input graph is about to run? If I input a graph that actually has finished, then this API call will still publish a graph started event, but this doesn't make sense.